File: e-commerce-backend/Models/Frontend/ProductModel.cs
using e_commerce_backend.Models.Backend;

namespace e_commerce_backend.Models.Frontend
{
    public class ProductModel
    {
        public string Id { get; set; } = string.Empty;
        public double CardPrice { get; set; } = 0;
        public double CommonPrice { get; set; } = 0;
        public required string Title { get; set; } = string.Empty;
        public int Rating { get; set; } = 0;
        public int Discount { get; set; } = 0;
        public required string Country { get; set; } = string.Empty;
        public required string Brand { get; set; } = string.Empty;
        public int Weight { get; set; } = 0;
        public int Articul { get; set; } = 0;
        public string ProductMain64Image { get; set; } = string.Empty;
        public required string ProductTag { get; set; } = string.Empty;
        public required string ProductCategory { get; set; } = string.Empty;
        public required List<Rewiev> Rewievs { get; set; }
    }
}

File: e-commerce-frontend/src/app/models/product.model.ts
export interface IProduct {
    id: string,
    cardPrice: number,
    commonPrice: number,
    title: string,
    country: string,
    rating: number,
    discount: number,
    brand: string,
    weight: number,
    articul: number,
    product64Images: string[],
    productMain64Image: string,
    productTag: ProductTag,
    productCategory: ProductCategory,
    rewievs: IRewiev[];
}

export enum ProductTag {
    none = 'none',
    news = 'news',
    sales = 'sales'
}

export enum ProductCategory {
    lactic = 'lactic',
    bread = 'bread',
    vegetables = 'vegetables',
    frozen = 'frozen',
    beverages = 'beverages',
    cakes = 'cakes',
    tea = 'tea',
    grocery = 'grocery',
    healthy = 'healthy',
    zoo = 'zoo',
    childish = 'childish',
    meat = 'meat',
    no_food = 'no_food'
}

export interface IRewiev {
    username: string;
    date: Date;
    rating: number;
    description?: string;
}
